src/libraries/System.Text.Json/src/System/Text/Json/JsonHelpers.Date.cs

```diff
@@ -211,11 +211,8 @@ private static bool TryParseDateTimeOffset(ReadOnlySpan<byte> source, out DateTi
         {
             parseData = default;
 
-            // Source does not have enough characters for YYYY-MM-DD
-            if (source.Length < 10)
-            {
-                return false;
-            }
+            // too short datetime
```

Comment:
I'll update this comment in a follow up. It should be something like 

"Validated upstream that the buffer has enough characters for YYYY-MM-DD"